### PR TITLE
Iterating reasons for rejection - new structure and interaction

### PR DESCRIPTION
**Pull request by adamsilver**

### CODE REVIEW

File: app/data/generators/rejection.js [hunk 1]
```diff
@@ -1,98 +1,70 @@
...
 
-  const behaviour = {
-    actions: 'Yes',
-    'actions-reasons': [
-      'Did not reply to messages',
-      'Other'
-    ],
-    'actions-reasons-other': faker.lorem.paragraph(),
-    'actions-reasons-other-improve': faker.lorem.paragraph()
+  let o = {
```
Comment (simonwhatley) [inline app/data/generators/rejection.js:6]: What is `o`? Can you use a more meaningful variable name?

File: app/data/generators/rejection.js [hunk 2]
```diff
@@ -1,98 +1,70 @@
...
-  }
+    ]
 
-  const interviewPerformance = {
-    'interview-performance': 'Yes',
-    'interview-performance-advice': faker.lorem.paragraph()
-  }
+    if(o['qualifications-reasons'].find(item => 'Degree does not meet course requirements')) {
+      o['qualifications-reasons-degree-does-not-meet-course-requirements'] = faker.lorem.paragraph(1)
+    }
```
Comment (simonwhatley) [inline app/data/generators/rejection.js:21]: Why do you need the if-statement if the item is always in the reasons array? Will you be randomising the reasons?

File: app/data/generators/rejection.js [hunk 3]
```diff
@@ -1,98 +1,70 @@
...
-  const course = {
-    'course-full': 'Yes'
-  }
 
-  const offeredOther = {
-    'other-offer': 'Yes',
-    'other-offer-details': faker.lorem.paragraph()
+    if(o['qualifications-reasons'].find(item => 'Other')) {
+      o['qualifications-reasons-other'] = faker.lorem.paragraph(1)
+    }
```
Comment (simonwhatley) [inline app/data/generators/rejection.js:26]: Why do you need the if-statement if the item is always in the reasons array? Will you be randomising the reasons?

File: app/data/generators/rejection.js [hunk 4]
```diff
@@ -1,98 +1,70 @@
...
-    ],
-    'safeguarding-reasons-vetting-information': faker.lorem.paragraph()
-  }
+    if(o['personal-statement-reasons'].find(item => 'Quality of writing')) {
+      o['personal-statement-reasons-quality-of-writing'] = faker.lorem.paragraph(1)
+    }
+
+    if(o['personal-statement-reasons'].find(item => 'Other')) {
+      o['personal-statement-reasons-other'] = faker.lorem.paragraph(1)
+    }
```
Comment (simonwhatley) [inline app/data/generators/rejection.js:44]: Same again. Why do you need the if-statement if the item is always in the reasons array? Will you be randomising the reasons?

File: app/views/_components/feedback/_feedback.scss [hunk 5]
```diff
@@ -1,6 +1,6 @@
 .app-feedback__section {
   position: relative;
-  padding-right: 70px;
+  // padding-right: 70px;
```
Comment (simonwhatley) [inline app/views/_components/feedback/_feedback.scss:3]: Delete rather than comment out?

File: app/views/_includes/candidate-feedback.njk [hunk 6]
```diff
@@ -1,314 +1,328 @@
...
-      <h2 class="govuk-heading-s">
-        Course full
-      </h2>
-      <p>The course you applied to was full.</p>
 
-      {% if rejectionChangeLinks %}
-        <p class="app-rejection-change-link"><a class="govuk-link govuk-link--no-visited-state" href="{{baseUrl}}">Change</a></p>
-      {% endif %}
+
+
```
Comment (simonwhatley) [inline app/views/_includes/candidate-feedback.njk:117]: I'll be your personal linter: unnecessary empty lines...

File: app/views/_includes/candidate-feedback.njk [hunk 7]
```diff
@@ -1,314 +1,328 @@
...
-        Visa application sponsorship
-      </h2>
 
-      <p>{{ rejectionReasons["sponsor-visa-details"] }}</p>
 
-      {% if rejectionChangeLinks %}
-        <p class="app-rejection-change-link"><a class="govuk-link govuk-link--no-visited-state" href="{{baseUrl}}">Change</a></p>
-      {% endif %}
+
+
```
Comment (simonwhatley) [inline app/views/_includes/candidate-feedback.njk:186]: Ahem! Unnecessary empty lines...

File: app/views/_includes/candidate-feedback.njk [hunk 8]
```diff
@@ -1,314 +1,328 @@
...
-      <h2 class="govuk-heading-s">
-        {{application["provider"]}} offered you a place on another course
-      </h2>
-      <p>{{rejectionReasons["other-offer-details"]}}</p>
 
-      {% if rejectionChangeLinks %}
-        <p class="app-rejection-change-link"><a class="govuk-link govuk-link--no-visited-state" href="{{baseUrl}}">Change</a></p>
-      {% endif %}
+
+
```
Comment (simonwhatley) [inline app/views/_includes/candidate-feedback.njk:231]: 🙈

File: app/views/_includes/candidate-feedback.njk [hunk 9]
```diff
@@ -1,314 +1,328 @@
...
-        {% endif %}
-
-      </ul>
-      {% if rejectionChangeLinks %}
-        <p class="app-rejection-change-link"><a class="govuk-link govuk-link--no-visited-state" href="{{baseUrl}}">Change</a></p>
-      {% endif %}
+
+
+
+
```
Comment (simonwhatley) [inline app/views/_includes/candidate-feedback.njk:251]: Oh the empty lines...

File: app/views/_includes/candidate-feedback.njk [hunk 10]
```diff
@@ -1,314 +1,328 @@
...
```
Comment (simonwhatley) [inline app/views/_includes/candidate-feedback.njk:271]: Empty lines

File: app/views/_includes/candidate-feedback.njk [hunk 11]
```diff
@@ -1,314 +1,328 @@
...
-        {{whyHeading}}
-      </h2>
-      <p>{{rejectionReasons['why']}}</p>
-      {% if rejectionChangeLinks %}
-        <p class="app-rejection-change-link"><a class="govuk-link govuk-link--no-visited-state" href="{{baseUrl}}/other-reasons-for-rejection/">Change</a></p>
-      {% endif %}
+
+
+
+
```
Comment (simonwhatley) [inline app/views/_includes/candidate-feedback.njk:291]: Unnecessary empty lines...

File: app/views/_includes/candidate-feedback.njk [hunk 12]
```diff
@@ -1,314 +1,328 @@
...
-        <h2 class="govuk-heading-s">
-          Additional feedback
-        </h2>
-        <p>{{rejectionReasons['other-feedback-details']}}</p>
-        {% if rejectionChangeLinks %}
-          <p class="app-rejection-change-link"><a class="govuk-link govuk-link--no-visited-state" href="{{baseUrl}}/other-reasons-for-rejection/">Change</a></p>
-        {% endif %}
+
+
+
```
Comment (simonwhatley) [inline app/views/_includes/candidate-feedback.njk:310]: Last one...

**Review by simonwhatley — CHANGES_REQUESTED** [review]

File: app/data/generators/rejection.js (hunk 2 above)
Comment (adamsilver) [reply app/data/generators/rejection.js]: Yes, this is a future proof thing. I just did the minimum to set up the structure. You cool with me mergin this?

File: app/data/generators/rejection.js (hunk 3 above)
Comment (adamsilver) [reply app/data/generators/rejection.js]: As above.

File: app/data/generators/rejection.js (hunk 4 above)
Comment (adamsilver) [reply app/data/generators/rejection.js]: As above

File: app/views/_components/feedback/_feedback.scss (hunk 5 above)
Comment (adamsilver) [reply app/views/_components/feedback/_feedback.scss:3]: Oh I should be cleaning this up completely. Thanks.

File: app/data/generators/rejection.js (hunk 2 above)
Comment (simonwhatley) [reply app/data/generators/rejection.js]: Yes, that's fine

File: app/data/generators/rejection.js (hunk 4 above)
Comment (simonwhatley) [reply app/data/generators/rejection.js]: Yes, that's fine

File: app/data/generators/rejection.js (hunk 3 above)
Comment (simonwhatley) [reply app/data/generators/rejection.js]: Yes, that's fine

**Review by simonwhatley — APPROVED** [review]

**Pull request merged**